### DISCUSSION TITLE
Does Visual Studio absolutly required to compile SuperSlicer ?

### DISCUSSION THREAD

**Discussion by DzzD** [supermerill/SuperSlicer#2804] [Q&A] [answered]
Visual Studio take about 15Go on my HDD and I only installed it to compile SuperSlicer.

I use VSCode as my IDE, is Visual Studio really required or can it be replaced by a simple subset of tools enabling SuperSlicer build & compilation ?

**Comment by supermerill** [comment]
I don't know.
The supported way i to use the vs 2019 c++ toolchain. If you can get it without the gui, it should be okay.
If you want to use another compiler (& toolchain), ask cmake for it, and be prepared to modify a good quantity of `#define`.

**Reply by DzzD** [reply]
I can compile through VS Code with the CMake extension, it works very well, I was just wondering If I really need the whole Visual Studio, but I have done some more tests and it seems to require only few component (3GO), I will post them for people interested.

**Comment by DzzD** [comment, marked as answer]
For thoses interrested to works on SuperSlicer with VSCode on Windows, here is a step by step guide : 
- Install CMake from https://cmake.org/install/ (after install, it may requiere to be added to the PATH variable)
- Install VSCode  https://code.visualstudio.com/ + after install, add the VSCode CMake extension
- Install Visual Studio Build Tools 2019

**You only require to enable the following checkboxes when installing Visual Studio Build Tools 2019 :** 
<img width="701" alt="image" src="https://user-images.githubusercontent.com/12198339/169798266-a7406391-a6a0-4732-b969-b4fa30decd40.png">

<img width="944" alt="image" src="https://user-images.githubusercontent.com/12198339/169803486-f9b166c8-c849-41c2-baed-73a44eb0b56c.png">


Then follow the official guide [How to compile on Windows](https://github.com/supermerill/SuperSlicer/blob/master/doc/How%20to%20build%20-%20Windows.md)

Note: you can replace the following : 
`cmake .. -G "Visual Studio 16 2019" -A x64 -DCMAKE_PREFIX_PATH="PATH_TO_Slic3r\deps\build\destdir\usr\local"`
`msbuild /m /P:Configuration=Release INSTALL.vcxproj`
by
`cmake .. -G "Visual Studio 16 2019" -A x64 -DCMAKE_PREFIX_PATH="%cd%\..\deps\build\destdir\usr\local"`
`msbuild /m /P:Configuration=Release INSTALL.vcxproj`

The first build take ages, but the subsequents are very fast.
After the first build you can then compile from within VSCode directly

**Comment by DzzD** [comment]
To enable the debug button you can add a file **./.vscode/launch.json**  containing :

```javascript
{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Super Slicer",
            "type": "cppvsdbg",
            "request": "launch",
            "program": "${workspaceFolder}/build/src/Debug/superslicer.exe",
            "args": [],
            "stopAtEntry": false,
            "cwd": "${fileDirname}",
            "environment": [],
            "console": "internalConsole"
        }
    ]
}

```

[launch.json.txt](https://github.com/supermerill/SuperSlicer/files/8754148/launch.json.txt)

File must be named launch.json... you need to remove the .txt extension...
